### PR TITLE
Refactored unit tests and integration tests

### PR DESCRIPTION
**Pull request by Rohan-cod**
## Description

Refactored unit tests (`test_unit.py`) and integration_tests (`test_integration.py`). Made functions for the code that was repetitively used in various tests. Also, replaced the `if-else` statements in `get_chrom` with a `switcher` using **[dictionary mapping](https://towardsdatascience.com/dictionary-as-an-alternative-to-if-else-76fe57a1e4af)**.

## How Has This Been Tested?

The code has been tested by running all the unit tests using the command `python -m unittest` and validated the PEP 8 formatting using `pycodestyle`.

- [x] Unit Tests
- [x] PEP 8 Formatting Validation

### CODE REVIEW

File: vcf2fhir/test/test_integration.py [hunk 1]
```diff
@@ -39,6 +39,173 @@ def _validate_phase_rel(fhir_json, map_variant_index):
...
+            actual_fhir_json, dict), True)
+        # Validate: list of observation uids and list of result uids same.
+        # Also set the uids to '' to avoid guid comparison in next step
+        map_ids = _get_uids_map(actual_fhir_json)
+        self.assertEqual(map_ids['obv_ids'], map_ids['result_ids'])
+        actual_fhir_json['issued'] = ''
+        # Finally, check if the acutal json after removing all uids is
+        # same as expected json
+        expected_fhir_json = json.load(expected_output_file)
+        self.assertEqual(actual_fhir_json, expected_fhir_json)
```
Comment (srgothi92) [inline vcf2fhir/test/test_integration.py:97]: Just keep this in this function. What you are trying to do is over refactoring. Your tests always need to be very simple and easy to follow. We would not prefer test helper methods to handle the actual test code for e.g `o_vcf_2_fhir = vcf2fhir.Converter(` with different argument combination makes a test.

File: vcf2fhir/test/test_integration.py [hunk 2]
```diff
@@ -39,6 +39,173 @@ def _validate_phase_rel(fhir_json, map_variant_index):
...
+
+
+def _test_empty_and_region_studied(
```
Comment (srgothi92) [inline vcf2fhir/test/test_integration.py:100]: Please revert this, if you start looking at actual tests you don't understand what you are testing.

File: vcf2fhir/test/test_integration.py [hunk 3]
```diff
@@ -39,6 +39,173 @@ def _validate_phase_rel(fhir_json, map_variant_index):
...
+        has_tabix=has_tabix,
+        conv_region_filename=conv_region_filename,
+        conv_region_dict=conv_region_dict,
+        region_studied_filename=region_studied_filename,
+        nocall_filename=nocall_filename)
+    o_vcf_2_fhir.convert(output_filename)
+
+
+def _compare_type_of_converter(
+        self,
```
Comment (srgothi92) [inline vcf2fhir/test/test_integration.py:136]: Please revert this, if you start looking at actual tests you don't understand what you are testing.

File: vcf2fhir/test/test_integration.py [hunk 4]
```diff
@@ -39,6 +39,173 @@ def _validate_phase_rel(fhir_json, map_variant_index):
...
+        has_tabix=has_tabix,
+        conv_region_filename=conv_region_filename,
+        conv_region_dict=conv_region_dict,
+        region_studied_filename=region_studied_filename,
+        nocall_filename=nocall_filename)
+    self.assertEqual(type(o_vcf_2_fhir), vcf2fhir.Converter)
+
+
+def _compare_exception_string(
+        self,
```
Comment (srgothi92) [inline vcf2fhir/test/test_integration.py:171]: Please revert this, if you start looking at actual tests you don't understand what you are testing.

**Review by srgothi92 — COMMENTED** [review]
Thanks for the effort, but I would have to ask you to revert lots of it. You always need to think about boundary between readability and number of lines of code. Sometimes it is okay to have repeated code when they look same but there are few parameters changing. Moving such code to functions would require you to add some complex if - else conditionals which makes it harder to read and follow. Important goal in tests is make it as simple as possible, your tests it self should not require another tests :)

**Comment by Rohan-cod** [comment]
@srgothi92, I have updated `test_integration.py` as per your suggestions and squashed the changes into a single commit. :)

File: vcf2fhir/test/test_unit.py [hunk 5]
```diff
@@ -6,96 +6,98 @@
 suite = doctest.DocTestSuite(vcf2fhir)
 
 
+def _test_utility_function(
```
Comment (srgothi92) [inline vcf2fhir/test/test_unit.py:9]: Why are we creating this method ? what existed before was good. Please revert this too.
Unrelated but the method name or signature does not convey anything about the method itself.

**Comment by Rohan-cod** [comment]
@srgothi92, I have updated `test_unit.py` as per your suggestions and squashed the changes into a single commit. :)

**Review by srgothi92 — APPROVED** [review]

**Pull request merged**